theseus/geometry/se2.py

```diff
@@ -92,37 +94,25 @@ def update_from_rot_and_trans(self, rotation: SO2, translation: Point2):
     # From https://github.com/strasdat/Sophus/blob/master/sophus/se2.hpp#L160
     def _log_map_impl(self) -> torch.Tensor:
         rotation = self.rotation
-
-        theta = rotation.log_map()
-        half_theta = 0.5 * theta.view(-1)
-
+        theta = rotation.log_map().view(-1)
         cosine, sine = rotation.to_cos_sin()
-        cos_minus_one = cosine - 1
-        halftheta_by_tan_of_halftheta = torch.zeros_like(cos_minus_one)
 
-        # Compute halftheta_by_tan_of_halftheta when theta is not near zero
-        idx_regular_vals = cos_minus_one.abs() > theseus.constants.EPS
-        halftheta_by_tan_of_halftheta[idx_regular_vals] = (
-            -(half_theta * sine)[idx_regular_vals] / cos_minus_one[idx_regular_vals]
+        # Compute the approximations when theta is near to 0
+        small_theta = theta.abs() < SE2.SE2_EPS
+        non_zero = torch.ones(1, dtype=self.dtype, device=self.device)
+        sine_nz = torch.where(small_theta, non_zero, sine)
+        a = (
```

Comment:
Probably give `a` and `b` a more descriptive name.

Reply:
Also since the method changed, does the reference need updating?

Reply:
I think both of these were addressed in [here](https://github.com/facebookresearch/theseus/pull/56/files#diff-0fb8d109b4f98eb8ec2a182c2e6a8d1a6817b9130517c2b03f56594f18063d86L91). However, I didn't see a new reference for these expressions, if there is none, then at least some explanation on where they come from would be useful.